app/lib/view-transition.tsx

```diff
@@ -0,0 +1,169 @@
+// requires that view-transitions and animations are setup in in the app's tailwind.css
+import {
+  Link as RemixLink,
+  useLocation,
+  useNavigate as useRemixNavigate,
+} from '@remix-run/react';
+import { create } from 'zustand';
+
+type AnimationDirection = 'left' | 'right' | 'up' | 'down';
+
+interface HistoryStackItem {
+  path: string;
+  /** direction of the transition that created this item */
+  direction: AnimationDirection;
+}
+
+interface TransitionHistoryStore {
+  history: HistoryStackItem[];
+  push: (entry: HistoryStackItem) => void;
+  pop: () => HistoryStackItem | undefined;
+  clear: () => void;
+}
+
+const useStore = create<TransitionHistoryStore>((set, get) => ({
```

Comment:
I don't like the complexity of maintaining our own navigation history. It also seems uncessary. WIll play with this a bit

Reply:
I think I found a better way see my branch `test-view-transition` and the code in `root.tsx` plus how I used Link component on the pages.

Demo is [here](https://boardwalk-git-test-view-transition-makeprisms.vercel.app/)

Reply:
Please let me know what you think now. I significantly changed `view-transition`. There is no more history state. We keep the ViewTransition component to limit boiler plate, add 'back' functionality, and make sure the transitions are used correctly. This can work as a NavLink or a Link.

I also made it so we can have dynamic animations. Default is static. See _protected._index for dynamic transitions